pkg/infra/iac2/templates/eks_node_group/factory.ts

```diff
@@ -0,0 +1,37 @@
+import * as aws from '@pulumi/aws'
+
+interface Args {
+    Name: string
+    Cluster: aws.eks.Cluster
+    NodeRole: aws.iam.Role
+    AmiType: string
+    Subnets: aws.ec2.Subnet[]
+    DesiredSize: number
+    MinSize: number
+    MaxSize: number
+    MaxUnavailable: number
+    DiskSize: number
+    InstanceTypes: string[]
+    Labels: Record<string, string>
+}
+
+// noinspection JSUnusedLocalSymbols
+function create(args: Args): aws.eks.NodeGroup {
+    return new aws.eks.NodeGroup(args.Name, {
+        clusterName: args.Cluster.name,
+        nodeRoleArn: args.NodeRole.arn,
+        amiType: args.AmiType != '' ? args.AmiType : undefined,
```

Comment:
It's too bad this go-ism (`""` being the null string) leaks into the output ts. I forget, do we have the ability to embed go-template rules directly into these ts files? If so, this could be a good place to add `/* TMPL if {{.AmiType}} */`. Otherwise, would you mind filing a ticket and adding it to the fit-and-finish milestone?

Reply:
yeah we do, i can add the templating here. We also want to port over the ami types so we support gpus which we dont have a ticket for so i will create that